megamek/src/megamek/client/ui/swing/PointblankShotDisplay.java

```diff
@@ -0,0 +1,1068 @@
+import java.awt.event.ItemListener;
+import java.util.ArrayList;
+import java.util.Arrays;
+import java.util.Hashtable;
+import java.util.Vector;
+
+import javax.swing.AbstractAction;
+import javax.swing.event.ListSelectionEvent;
+import javax.swing.event.ListSelectionListener;
+
+import megamek.client.event.BoardViewEvent;
+import megamek.client.ui.Messages;
+import megamek.client.ui.swing.util.CommandAction;
+import megamek.client.ui.swing.util.KeyCommandBind;
+import megamek.client.ui.swing.util.MegaMekController;
+import megamek.client.ui.swing.widget.MegamekButton;
+import megamek.common.AmmoType;
+import megamek.common.Compute;
+import megamek.common.Coords;
+import megamek.common.Entity;
+import megamek.common.HexTarget;
+import megamek.common.IAimingModes;
+import megamek.common.IBoard;
+import megamek.common.IGame;
+import megamek.common.Mounted;
+import megamek.common.TargetRoll;
+import megamek.common.Targetable;
+import megamek.common.ToHitData;
+import megamek.common.WeaponType;
+import megamek.common.actions.ArtilleryAttackAction;
+import megamek.common.actions.EntityAction;
+import megamek.common.actions.WeaponAttackAction;
+import megamek.common.event.GamePhaseChangeEvent;
+import megamek.common.event.GameTurnChangeEvent;
+import megamek.common.options.OptionsConstants;
+
+/**
+ * This display is used for when hidden units are taking pointblank shots.
+ * 
+ * @author arlith
+ *
+ */
+public class PointblankShotDisplay extends FiringDisplay implements
+        ItemListener, ListSelectionListener {
+
+    /**
+     * 
+     */
+    private static final long serialVersionUID = -58785096133753153L;
+
+    /**
+     * This enumeration lists all of the possible ActionCommands that can be
+     * carried out during the pointblank phase.  Each command has a string for
+     * the command plus a flag that determines what unit type it is
+     * appropriate for.
+     *
+     * @author arlith
+     */
+    public static enum FiringCommand implements PhaseCommand {
+        FIRE_TWIST("fireTwist"),
+        FIRE_FIRE("fireFire"),
+        FIRE_SKIP("fireSkip"),
+        FIRE_MODE("fireMode"),
+        FIRE_FLIP_ARMS("fireFlipArms"),
+        FIRE_SEARCHLIGHT("fireSearchlight"),
+        FIRE_CALLED("fireCalled"),
+        FIRE_CANCEL("fireCancel"),
+        FIRE_MORE("fireMore");
+
+        String cmd;
+
+        /**
+         * Priority that determines this buttons order
+         */
+        public int priority;
+
+        private FiringCommand(String c) {
+            cmd = c;
+        }
+
+        public String getCmd() {
+            return cmd;
+        }
+
+        public int getPriority() {
+            return priority;
+        }
+
+        public void setPriority(int p) {
+            priority = p;
+        }
+
+        public String toString() {
+            return Messages.getString("FiringDisplay." + getCmd());
+        }
+    }
+
+    // buttons
+    protected Hashtable<FiringCommand, MegamekButton> buttons;
```

Comment:
It should be a Map.  This was copied from `FiringDisplay` code, which was originally written as a `Hashtable`, from a time when I was less careful about types.  All of the *Display classes should have their buttons `Hashtables` changed to `Map`.